### PR TITLE
Handle cctor dependency on RVA static access

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Contributes to #68278.

Cc @dotnet/ilc-contrib

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs [hunk 1]
```diff
@@ -1045,7 +1045,9 @@ private void ImportFieldAccess(int token, bool isStatic, string reason)
                     // magic fields the compiler synthetized, the data blob might bring more dependencies
                     // and we need to scan those.
                     _dependencies.Add(_compilation.GetFieldRvaData(field), reason);
-                    // TODO: lazy cctor dependency
+                    Debug.Assert(!field.OwningType.HasInstantiation);
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs:1048]: Are fields with RVAs disallowed in ECMA or rejected by CoreCLR? Or is this an unimplemented case?

Comment (MichalStrehovsky) [reply src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs:1048]: ECMA doesn't seem to disallow it but this code will crash if it's on a generic type; we need double the lines to handle that.

These would map to the same thing irrespective of what's the T and that feels weird enough that I didn't feel like writing handling/tests for it.

Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs:1048]:
```suggestion
                    // RVA static fields in generic types not implemented
                    Debug.Assert(!field.OwningType.HasInstantiation);
```

**Review by jkotas — APPROVED** [review]

**Pull request merged**